_tools/types/main.go

```diff
@@ -0,0 +1,341 @@
+// Generates a report about every driver wich UASTv2 types does it use,
+// both in actual test fixtures and though the source code.
+package main
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"reflect"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/bblfsh/sdk/driver/manifest/discovery"
+	"github.com/bblfsh/sdk/uast"
+
+	_ "net/http/pprof"
+)
+
+const (
+	fixtureDir = "fixtures"
+	fixtureExt = ".sem.uast"
+	goDocURL   = "https://godoc.org/github.com/bblfsh/sdk/uast#"
+)
+
+var (
+	reposRootPath = path.Join(".", "drivers")
+	uastNodeRe    = regexp.MustCompile("uast:[^\"]*")
+
+	pprof      = flag.Bool("pprof", false, "start pprof profiler http endpoing")
+	skipUpdate = flag.Bool("skip", false, "skip git clone or pull")
+)
+
+func main() {
+	flag.Parse()
+
+	if *pprof {
+		pprofAddr := "localhost:6060"
+		fmt.Fprintf(os.Stderr, "running pprof on %s\n", pprofAddr)
+		go func() {
+			if err := http.ListenAndServe(pprofAddr, nil); err != nil {
+				log.Fatal("cannot start pprof: %v", err)
+			}
+		}()
+	}
+
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run() error {
+	drivers, err := listDrivers()
+	if err != nil {
+		return fmt.Errorf("failed to list drivers: %s", err)
+	}
+
+	if !*skipUpdate {
+		err = maybeCloneOrPullAll(drivers)
+		if err != nil {
+			return fmt.Errorf("failed to pull driver repos: %s", err)
+		}
+	}
+
+	uastTypes := findUASTTypesInSDK()
+	for _, driver := range drivers {
+		if err := analyzeFixtures(driver); err != nil {
+			return err
+		}
+		analyzeCode(driver, uastTypes)
+	}
+
+	formatMarkdownTable(drivers, uastTypes)
+	return nil
+}
+
+type driverStats struct {
+	url          string
+	language     string
+	path         string
+	skip         bool
+	fixturesNum  int
+	fixturesUast map[string]int
+	codeUast     map[string]int
+}
+
+// listDrivers lists all available drivers.
+func listDrivers() ([]*driverStats, error) {
+	fmt.Fprintf(os.Stderr, "discovering all available drivers\n")
```

Comment:
Minor nit:

```suggestion
	fmt.Fprintf(os.Stderr, "Discovering all available drivers ...\n")
```

Reply:
This script uses the convention of not capitalizing anything. Please let me know if you have a strong preferences agains and then I'll change it everywhere for consistency.

Otherwise going to keep it as is.

Reply:
It's purely an aesthetic recommendation, that's fine.